kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java

```diff
@@ -787,6 +790,37 @@ protected EntityManagerFactory build(Map<String, String> properties) {
         }
     }
 
+    private void checkAndAddTaskAssigningEntities(PersistenceUnitInfo info) {
```

Comment:
What is the purpose of this method? Is it required to register task planning JPA entities into the PersistenceContext to make them writeable into database together with jBPM classes?

In other words, does it serve the purpose of saving the task assignment at a safe point in a jBPM process?

Reply:
yes, tasks assigning incorporates one entity that we need to have in the same persistence unit as the jBPM ones. But I don't want to add it to the core jBPM entities to not add this such specific stuff there. So basically this entity is incorporated only on servers that has the tasks assigning enabled.

Reply:
Would you please point me to a piece of code which effectively saves the planning model in DB?

I doubt we should leak the task assignment implementation classes into jBPM KIE extension like here.

Reply:
sure, see here

https://github.com/kiegroup/droolsjbpm-integration/blob/842cdd76a6181d4a11da8afd8aab8d14e1df92f7/jbpm-task-assigning/kie-server-services-task-assigning/kie-server-services-task-assigning-runtime/src/main/java/org/kie/server/services/taskassigning/runtime/TaskAssigningRuntimeServiceBase.java#L389

Reply:
Thanks, let's talk about it in today's meeting - I don't think we should couple the persistence of jBPM and task assignment so closely.